internal/vm/vsock.go

```diff
@@ -240,35 +238,31 @@ func tryAccept(logger *logrus.Entry, listener net.Listener, port uint32) (net.Co
 		}
 	}()
 
-	err = tryConnWrite(conn, vsockAckMsg(port), vsockAckMsgTimeout)
-	if err != nil {
-		return nil, vsockAckError{cause: err}
-	}
-
 	return conn, nil
 }
 
-// tryConnRead will try to do a read from the provided conn, returning an error if
-// the bytes read does not match what was provided or if the read does not complete
+// tryConnReadUntil will try to do a read from the provided conn till encounter
+// the specified end character, returning an error if the read does not complete
 // within the provided timeout. It will reset socket deadlines to none after returning.
 // It's only intended to be used for connect/ack messages, not general purpose reads
 // after the vsock connection is established fully.
-func tryConnRead(conn net.Conn, expectedRead string, timeout time.Duration) error {
+func tryConnReadUntil(conn net.Conn, end byte, timeout time.Duration) ([]byte, error) {
 	conn.SetDeadline(time.Now().Add(timeout))
 	defer conn.SetDeadline(time.Time{})
 
-	actualRead := make([]byte, len(expectedRead))
-	_, err := conn.Read(actualRead)
-	if err != nil {
-		return err
-	}
-
-	if expectedRead != string(actualRead) {
-		return errors.Errorf("expected to read %q, but instead read %q",
-			expectedRead, string(actualRead))
+	var line []byte
+	for {
+		var c [1]byte
+		_, err := conn.Read(c[:])
+		if err != nil {
+			return nil, err
+		}
+		line = append(line, c[0])
```

Comment:
can be just `return bufio.NewReaderSize(conn, 32).ReadSlice('\n')` ?

Or even `...ReadString('\n')` since you later cast  it to string

Reply:
Oh, nice. Will change.

Reply:
Done.